blueocean-dashboard/src/main/js/components/StageRestartLink.tsx

```diff
@@ -0,0 +1,88 @@
+import * as React from 'react';
+import { RunApi as runApi } from '@jenkins-cd/blueocean-core-js';
+import { ToastUtils } from '@jenkins-cd/blueocean-core-js';
+
+interface Props {
+    title: string;
+    run: object;
+    nodeRestartId: number;
+    pipeline: Pipeline;
+    onNavigation: Function;
+    t: Function;
+}
+
+interface State {
+    replaying: boolean;
+}
+
+interface Pipeline {
+    name: string;
+}
+
+class StageRestartLink extends React.Component<Props, State> {
+    constructor(props) {
+        super(props);
+        this.state = {
+            replaying: false,
+        };
+    }
+
+    _onRestartStageClick() {
+        if (this.state.replaying) {
+            return;
+        }
+
+        this.setState({
+            replaying: true,
+        });
+
+        //change run to latest Run
+        runApi
+            .restartStage(this.props.run, this.props.nodeRestartId)
+            .then(run => ToastUtils.createRunStartedToast(this.props.pipeline, run, this.props.onNavigation))
+            .then(runDetailsUrl => this._afterReplayStarted(runDetailsUrl));
+    }
+
+    _afterReplayStarted(runDetailsUrl) {
+        this.props.onNavigation(runDetailsUrl);
+    }
+
+    restartStageIcon() {
+        return (
+            <svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
```

Comment:
I think this icon should be a separate reaction component, perhaps a functional one. Maybe something like,  ‘<RestartStageIcon />’.